src/transformers/models/beit3/modeling_beit3.py

```diff
@@ -0,0 +1,1460 @@
+
+    def get_num_layers(self):
+        return self.encoder.num_layers
+
+    @add_start_docstrings_to_model_forward(BEIT3_MODEL)
+    @replace_return_docstrings(output_type=BaseModelOutputWithPooling, config_class=_CONFIG_FOR_DOC)
+    def forward(
+        self,
+        input_ids=None,
+        pixel_values=None,
+        attention_mask=None,
+        image_text_mask=None,
+        vision_masked_position=None,
+        past_key_value=None,
+        output_hidden_states=None,
+        output_attentions=None,
+        return_dict=None,
+    ):
+        r"""
+        Returns:
+
+        Examples:
+
+        ```python
+        >>> from transformers import Beit3Processor, Beit3Model
+        >>> from PIL import Image
+        >>> import requests
+
+        >>> processor = Beit3Processor.from_pretrained("Raghavan/beit3_base_patch16_224_in1k")
+        >>> model = Beit3Model.from_pretrained("Raghavan/beit3_base_patch16_224_in1k")
+
+        >>> url = "http://images.cocodataset.org/val2017/000000039769.jpg"
+        >>> image = Image.open(requests.get(url, stream=True).raw)
+        >>> text = "This is photo of a cat"
+
+        >>> inputs = processor(text=text, images=image, return_tensors="pt")
+
+        >>> # forward pass
+        >>> outputs = model(**inputs)
+
+        >>> # last hidden states have a sequence length equal to
+        >>> # the number of text tokens + number of image patches + 1 (for the CLS token)
+        >>> # which in this case equals 8 + (224//16) + 1 = 205
+        >>> last_hidden_state = outputs.last_hidden_state
+        >>> print(last_hidden_state.shape)
+        torch.Size([1, 205, 768])
+        ```"""
+
+        output_attentions = output_attentions if output_attentions is not None else self.config.output_attentions
+        output_hidden_states = (
+            output_hidden_states if output_hidden_states is not None else self.config.output_hidden_states
+        )
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+
+        text_end_positions = None
+        if input_ids is None and pixel_values is None:
+            raise ValueError("You have to specify at least input_ids or pixel_values")
+
+        if input_ids is None:
+            embeddings = self.vision_embedding(pixel_values, vision_masked_position)
+            multiway_split_position = -1
+        elif pixel_values is None:
+            embeddings = self.text_embedding(input_ids)
+            multiway_split_position = 0
+        else:
+            vision_embeddings = self.vision_embedding(pixel_values, vision_masked_position)
+            multiway_split_position = vision_embeddings.size(1)
+            text_embeddings = self.text_embedding(input_ids)
+            embeddings = torch.cat([vision_embeddings, text_embeddings], dim=1)
+
+            if attention_mask is not None:
+                zeros_for_vision_padding = torch.ones(vision_embeddings.shape[:-1]).to(vision_embeddings.device).bool()
```

Comment:
There should be a comment here saying what shape `zeros_for_vision_padding` is

Reply:
Done.